crypto/keyring/keyring.go

```diff
@@ -426,6 +429,31 @@ func (ks keystore) DeleteByAddress(address sdk.Address) error {
 	return nil
 }
 
+func (ks keystore) Rename(from, to string) error {
+	_, err := ks.Key(to)
+	if err == nil {
+		return errors.New(fmt.Sprintf("rename failed: %s already exists in keyring", to))
+	}
+
+	passPhrase := "temp"
+	armor, err := ks.ExportPrivKeyArmor(from, passPhrase)
+	if err != nil {
+		return err
+	}
+
+	err = ks.Delete(from)
```

Comment:
```suggestion
	err = ks.Delete(oldName)
```

Reply:
moving delete to the last step would cause the import to error, as it doesn't allow duplicate pubkeys to the keyring. 

perhaps we can 

1) create a dump file when errors occur with the key information so the user can manually import on errors. 

or 

2) (not sure if possible) see if we can accept duplicate pubkeys in the keyring